test/csuite/parse_opts/main.c

```diff
@@ -0,0 +1,315 @@
+    REPORT_INT(opts, table_type);
+    REPORT_INT(opts, do_data_ops);
+    REPORT_INT(opts, preserve);
+    REPORT_INT(opts, tiered_storage);
+    REPORT_INT(opts, verbose);
+    REPORT_INT(opts, nrecords);
+    REPORT_INT(opts, nops);
+    REPORT_INT(opts, nthreads);
+    REPORT_INT(opts, n_append_threads);
+    REPORT_INT(opts, n_read_threads);
+    REPORT_INT(opts, n_write_threads);
+    REPORT_STR(x_opts, c_option);
+    REPORT_INT(x_opts, d_option);
+    REPORT_INT(x_opts, e_option);
+    REPORT_INT(x_opts, f_option);
+}
+
+/*
+ * check --
+ *     Call testutil_parse_opts and return options.
+ */
+static void
+check(int argc, char *const *argv, TEST_OPTS *opts, EXTENDED_OPTS *x_opts)
+{
+    int ch;
+    const char *prog;
+    const char *x_usage = " [-c string] [-d] [-e] [-f int]";
+
+    memset(opts, 0, sizeof(*opts));
+    memset(x_opts, 0, sizeof(*x_opts));
+
+    /* This may be called multiple times, so reset the __wt_getopt parser. */
+    __wt_optind = 1;
+    __wt_optreset = 1;
+
+    prog = argv[0];
+    if (strchr(prog, '/') != NULL)
+        prog = strchr(prog, '/') + 1;
+
+    if (strcmp(prog, "parse_opts") == 0) {
+        /* Regular test of testutil_parse_opts, using only the options that it provides. */
+
+        testutil_check(testutil_parse_opts(argc, argv, opts));
+    } else {
+        /*
+         * Test of extended parsing, in which we'll parse some options that we know about and rely
+         * on testutil_parse_opt to cover the options it knows about.
+         */
+        testutil_assert(strcmp(prog, "parse_opt") == 0);
+
+        /*
+         * For this part of the testing, we're extending the list of options we're parsing, and
+         * using testutil_parse_opt to parse a subset of the standard options.
+         */
+        testutil_parse_opt_begin(argc, argv, "b:P:T:v", opts);
+        while ((ch = __wt_getopt(opts->progname, argc, argv, "b:c:def:P:T:v")) != EOF)
+            switch (ch) {
+            case 'c':
+                x_opts->c_option = __wt_optarg;
+                break;
+            case 'd':
+                x_opts->d_option = true;
+                break;
+            case 'e':
+                x_opts->e_option = true;
+                break;
+            case 'f':
+                x_opts->f_option = atoi(__wt_optarg);
+                break;
+            default:
+                if (testutil_parse_opt(opts, ch) != 0) {
+                    (void)fprintf(stderr, "usage: %s%s%s\n", opts->progname, x_usage, opts->usage);
+                    testutil_assert(false);
+                }
+            }
+        testutil_parse_opt_end(opts);
+    }
+}
+
+/*
+ * verify_expect --
+ *     Verify the returned options against the expected options.
+ */
+static void
+verify_expect(TEST_OPTS *opts, EXTENDED_OPTS *x_opts, TEST_OPTS *expect, EXTENDED_OPTS *x_expect)
+{
+#define VERIFY_INT(o, e, field)         \
+    if (o->field != 0 || e->field != 0) \
+    testutil_assert(o->field == e->field)
```

Comment:
Same here about the indentation (although the VERIFY_STR below does not have this problem).

Reply:
Fixed.